src/connectors/hits/connectHits.js

```diff
@@ -48,23 +70,126 @@ Full documentation available at https://community.algolia.com/instantsearch.js/c
 export default function connectHits(renderFn) {
   checkRendering(renderFn, usage);
 
-  return (widgetParams = {}) => ({
-    init({instantSearchInstance}) {
-      renderFn({
-        hits: [],
-        results: undefined,
-        instantSearchInstance,
-        widgetParams,
-      }, true);
-    },
+  return (widgetParams = {}) => {
+    const {
+      escapeHits = false,
+      escapeHitsWhitelist = [],
+    } = widgetParams;
+
+    if (!isArray(escapeHitsWhitelist)) {
+      throw new Error(usage);
+    }
+
+    return {
+      getConfiguration() {
+        return config;
+      },
+
+      init({instantSearchInstance}) {
+        renderFn({
+          hits: [],
+          results: undefined,
+          instantSearchInstance,
+          widgetParams,
+        }, true);
+      },
+
+      render({results, instantSearchInstance}) {
+        if (escapeHits === true) {
+          results.hits = recursiveEscape(results.hits, escapeHitsWhitelist);
+        }
+
+        renderFn({
+          hits: results.hits,
+          results,
+          instantSearchInstance,
+          widgetParams,
+        }, false);
+      },
+    };
+  };
+}
+
+function replaceWithEm(val) {
+  if (isPlainObject(val)) {
+    return reduce(
+      val,
+      (result, value, key) => {
+        result[key] = replaceWithEm(value);
+        return {};
```

Comment:
Why does it return an empty object here? Does it mean that we could replace `reduce` with a `forEach`?